src/palace/manager/api/admin/controller/report.py

```diff
@@ -46,10 +56,65 @@ def _authorize_from_request(
 
 
 class ReportController(LoggerMixin):
+
     def __init__(self, db: Session, registry: LicenseProvidersRegistry):
         self._db = db
         self.registry = registry
 
+    @classmethod
+    def report_for_key(cls, key: str) -> type[LibraryCollectionReport]:
+        if key not in REPORT_KEY_MAPPING:
+            detail = INVALID_REPORT_KEY.detail or "Unknown report key."
+            raise ProblemDetailException(
+                INVALID_REPORT_KEY.detailed(f"{detail.rstrip('. ')} (key='{key}').")
+            )
+        return REPORT_KEY_MAPPING[key]
+
+    def generate_report(self, *, report_key: str) -> Response:
+        """Generate the report indicated by the report_key."""
+
+        admin, library = _authorize_from_request(flask.request)
+        email_address = admin.email
+
+        report = self.report_for_key(report_key)
+        report_title = report.TITLE
+
+        request_id = uuid_encode(uuid.uuid4())
+
+        self.log.info(
+            f"Report '{report_title}' ({report_key}) requested by <{email_address}>. (request ID: {request_id})"
+        )
+        try:
+            task = generate_report.delay(
+                key=report_key,
+                request_id=request_id,
+                library_id=library.id,
+                email_address=email_address,
+            )
+        except Exception as e:
+            msg = f"Failed to generate report '{report_title}' ({report_key}). (request ID: {request_id})"
+            self.log.error(msg=msg, exc_info=e)
+            self._db.rollback()
```

Comment:
Minor: Is this rollback actually necessary here? It doesn't look like we should have done anything that needs a rollback.

Reply:
No, it is not. Good catch. 🎉